### ISSUE TITLE
A new lint for an assignment statement `a = b = c`

### ISSUE THREAD

**Issue by fiveseven-lambda** [rust-lang/rust-clippy#6576]
### What it does

regarding to: https://stackoverflow.com/questions/65661530/why-can-assignment-operator-be-chained-in-rust

One might assume an assignment statement `a = b = c;` to be equivalent to `a = c; b = c;`, but it isn't.

There seems to be very few cases, if any, where we would intentionally write `a = b = c;`.

It would be nice if clippy warned such an assignment.

### Categories (optional)

- Kind: correctness

### Drawbacks

None.

### Example

```rust
a = b = c;
```

Could be written as:

```rust
b = c;
a = ();
```

Or the programmer might have intended:

```rust
a = c;
b = c;
```

**Comment by camsteffen** [comment]
In practice, I think the programmer will just run into a type error. ([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d333ca42de981d8175169139a10d3c18))

**Comment by fiveseven-lambda** [comment]
@camsteffen  You used `Vec::with_capacity` in your example function, but if you only try debug print like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f102db5660fe6932e5e6f0bd0d8fb00c), it compiles with no warning about `a = b = c;`.
